src/test/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkTaskTest.java

```diff
@@ -177,6 +174,49 @@ public void closeTaskAndWaitToFinish() {
         assertTrue(l3-l2 > sleepTime  && l3-l2 < sleepTime + 1000);
     }
 
+    @Test
+    public void testCreateKustoEngineClient(){
+
+        HashMap<String, String> configs = KustoSinkConnectorConfigTest.setupConfigs();
+        configs.put(KustoSinkConfig.KUSTO_SINK_FLUSH_INTERVAL_MS_CONF, "100");
+        KustoSinkTask kustoSinkTask = new KustoSinkTask();
+        KustoSinkTask kustoSinkTaskSpy = spy(kustoSinkTask);
+        doNothing().when(kustoSinkTaskSpy).validateTableMappings(Mockito.<KustoSinkConfig>any());
+        kustoSinkTaskSpy.start(configs);
+
+
+        KustoSinkConfig kustoSinkConfig = new KustoSinkConfig(configs);
+
+
+        Client client = KustoSinkTask.createKustoEngineClient(kustoSinkConfig);
+        Assertions.assertNotNull(client);
+
+    }
+
+    @Test
+    public void testValidateTableMappingsConeectError(){
+        HashMap<String, String> configs = KustoSinkConnectorConfigTest.setupConfigs();
+        configs.put(KustoSinkConfig.KUSTO_SINK_FLUSH_INTERVAL_MS_CONF, "100");
+        KustoSinkTask kustoSinkTask = new KustoSinkTask();
+        KustoSinkTask kustoSinkTaskSpy = spy(kustoSinkTask);
+        doNothing().when(kustoSinkTaskSpy).validateTableMappings(Mockito.<KustoSinkConfig>any());
+        kustoSinkTaskSpy.start(configs);
+
+        KustoSinkConfig kustoSinkConfig = new KustoSinkConfig(configs);
+
+        assertThrows(ConnectException.class,()->{kustoSinkTask.validateTableMappings(kustoSinkConfig);});
+
+    }
+
+
+
+
+    @Test
+    public void testStop(){
+        KustoSinkTask kustoSinkTask = new KustoSinkTask();
```

Comment:
test with no assertions , please add what is the expected side effect of calling stop

Reply:
added Mockito.verify to test the void method

Reply:
@tanmaya-panda1 This is an incomplete test

If we remove the full body of stop in KustoSinkTask , and run this test , it will still pass

A good test will test the behavior and cover interactions.  Here is an example (hacked up not tested, but gives you the idea overall)

We can see that the behavior of close() is to close all the topic writers and also close the ingest client.  This is the success case , similarly in case of the IOException , the method should not throw , but rather just logs the exception and ends silently , this is a new test case for failure.

Similar is the case for the start() as well, else the tests will cover lines with absolutely no way of covering functionality , side effects etc.  There is not a lot of benefits when you do regressions if the tests are incomplete.

```
    @Test
    public void testStopSuccess() throws IOException {
        //set-up

        // easy to set it this way than mock
        TopicPartition mockPartition = new TopicPartition("test-topic",0);
        TopicPartitionWriter mockPartitionWriter = mock(TopicPartitionWriter.class);
        doNothing().when(mockPartitionWriter).close();
        IngestClient mockClient = mock(IngestClient.class);
        doNothing().when(mockClient).close();
        KustoSinkTask kustoSinkTask = new KustoSinkTask();
        // There is no mutate constructor
        kustoSinkTask.writers = Collections.singletonMap(mockPartition,mockPartitionWriter);
        kustoSinkTask.kustoIngestClient = mockClient;

        // when
        kustoSinkTask.stop();

        // then
        verify(mockClient,times(1)).close();
        verify(mockPartitionWriter,times(1)).close();
    }

```

Reply:
Got it @ag-ramachandran , added tests for stop on failure as well

Reply:
@tanmaya-panda1  Thanks , please make similar checks for other test cases as well